cmd/clusterctl/pkg/client/repository/components_client_test.go

```diff
@@ -0,0 +1,242 @@
+				configVariablesClient: test.NewFakeVariableClient().WithVar(variableName, variableValue),
+			},
+			args: args{
+				version:           "v1.0.0",
+				targetNamespace:   "ns2",
+				watchingNamespace: "",
+			},
+			want: want{
+				provider:          p1,
+				version:           "v1.0.0", // version detected
+				targetNamespace:   "ns2",    // target targetNamespace override default targetNamespace
+				watchingNamespace: "",
+				variables:         []string{variableName}, // variable detected
+			},
+			wantErr: false,
+		},
+		{
+			name: "Fails if components file does not exists",
+			fields: fields{
+				provider: p1,
+				repository: test.NewFakeRepository().
+					WithPaths("root", "components.yaml").
+					WithDefaultVersion("v1.0.0"),
+				configVariablesClient: test.NewFakeVariableClient().WithVar(variableName, variableValue),
+			},
+			args: args{
+				version:           "v1.0.0",
+				targetNamespace:   "",
+				watchingNamespace: "",
+			},
+			wantErr: true,
+		},
+		{
+			name: "Fails if default targetNamespace does not exists",
+			fields: fields{
+				provider: p1,
+				repository: test.NewFakeRepository().
+					WithPaths("root", "components.yaml").
+					WithDefaultVersion("v1.0.0").
+					WithFile("v1.0.0", "components.yaml", util.JoinYaml(podYaml, configMapYaml)),
+				configVariablesClient: test.NewFakeVariableClient().WithVar(variableName, variableValue),
+			},
+			args: args{
+				version:           "v1.0.0",
+				targetNamespace:   "",
+				watchingNamespace: "",
+			},
+			wantErr: true,
+		},
+		{
+			name: "Pass if default targetNamespace does not exists but a target targetNamespace is set",
+			fields: fields{
+				provider: p1,
+				repository: test.NewFakeRepository().
+					WithPaths("root", "components.yaml").
+					WithDefaultVersion("v1.0.0").
+					WithFile("v1.0.0", "components.yaml", util.JoinYaml(podYaml, configMapYaml)),
+				configVariablesClient: test.NewFakeVariableClient().WithVar(variableName, variableValue),
+			},
+			args: args{
+				version:           "v1.0.0",
+				targetNamespace:   "ns2",
+				watchingNamespace: "",
+			},
+			want: want{
+				provider:          p1,
+				version:           "v1.0.0", // version detected
+				targetNamespace:   "ns2",    // target targetNamespace applied
+				watchingNamespace: "",
+				variables:         []string{variableName}, // variable detected
+			},
+			wantErr: false,
+		},
+		{
+			name: "Fails if requested version does not exists",
+			fields: fields{
+				provider: p1,
+				repository: test.NewFakeRepository().
+					WithPaths("root", "components.yaml").
+					WithDefaultVersion("v1.0.0").
+					WithFile("v1.0.0", "components.yaml", util.JoinYaml(podYaml, configMapYaml)),
+				configVariablesClient: test.NewFakeVariableClient().WithVar(variableName, variableValue),
+			},
+			args: args{
+				version:           "v2.0.0",
+				targetNamespace:   "",
+				watchingNamespace: "",
+			},
+			wantErr: true,
+		},
```

Comment:
nit: consider adding test cases for `watchingNamespace` similar to `targetNamespace`

Reply:
done